appinventor/appengine/src/com/google/appinventor/client/explorer/dialogs/ProjectPropertiesDialogBox.java

```diff
@@ -0,0 +1,192 @@
+
+    /**
+     * Show the project property dialog
+     * 
+     * @param screenName name of the screen in which dialog needs to open
+     */
+    public void showDialog(String screenName) {
+        currentScreen = screenName;
+        Scheduler.get().scheduleDeferred(new Scheduler.ScheduledCommand() {
+            @Override
+            public void execute() {
+                projectProperties.center();
+            }
+        });   
+    }
+
+    public ProjectPropertiesDialogBox() {
+        uiBinder.createAndBindUi(this);
+        projectProperties.setAutoHideEnabled(false);
+        projectProperties.setModal(true);
+        projectProperties.setCaption(Ode.MESSAGES.projectPropertiesText());
+
+        // Get current instance of YaProjectEditor
+    	projectEditor = (YaProjectEditor)Ode.getInstance().getEditorManager().getOpenProjectEditor(
+            Ode.getInstance().getCurrentYoungAndroidProjectId());
+
+        MockForm form = projectEditor.getFormFileEditor("Screen1").getForm();
+
+        // Get project properties from the screen1 MockForm
+	    EditableProperties editableProperties = form.getProperties();
+        Iterator<EditableProperty> properties = editableProperties.iterator();
+
+        // Iterate and put the editable property to the corresponding category in categoryToProperties
+        while (properties.hasNext()) {
+            EditableProperty property = properties.next();
+
+            if (!categoryToProperties.containsKey(property.getCategory())) {
+                categoryToProperties.put(property.getCategory(), new ArrayList<EditableProperty>());
+            } 
+
+            categoryToProperties.get(property.getCategory()).add(property);
+        }
+
+        // Add the Categories to ListBox - projectPropertyCategoryTitlePanel
+        for (String categoryTitle : projectPropertyCategoryTitle) {
+            projectPropertyCategoryTitlePanel.addItem(categoryTitle);
+            propertiesDeckPanel.add(getPanel(categoryTitle));
+        }
+
+        // When category is changed by the user, display related properties
+        projectPropertyCategoryTitlePanel.addChangeHandler(new ChangeHandler() {
+            @Override
+            public void onChange(ChangeEvent event) {
+                propertiesDeckPanel.showWidget(projectPropertyCategoryTitlePanel.getSelectedIndex());
+            }
+        });
+
+        projectPropertyCategoryTitlePanel.setVisibleItemCount(projectPropertyCategoryTitle.size());
+        
+        // When dialog is opened, properties related to the General category is shown
+        propertiesDeckPanel.showWidget(0);
+    }
+
+    /**
+     * Build vertical panel for each categories
+     * 
+     * @param category indicates the category for which we need to build the vertical panel
+     * @return vertical panel which contains the all the Editable Property belongs to the particualt category passed as argument
+     */
+    private VerticalPanel getPanel(String category) {
```

Comment:
In the new UI we prefer to use FlowPanel (which maps to a `<div>` element) rather than VerticalPanel (which maps to a `<table>` element).